### PR TITLE
Fix ffmpeg compilation for rtd1296 architecture

### PR DESCRIPTION
**Pull request by m4tt075**
_Motivation:_ `ffmpeg` compilation has been broken for `rtd1296` platform and all packages depending on `ffmpeg`. This PR fixes those issue. `x264` has been upgraded to the most recent stable snapshot in the process as well. Untested so far, as I do not own a `rtd1296` device.

### Checklist
- [x] Build rule completed successfully for `rtd1296` architecture
- [ ] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [ ] New installation of package completed successfully

### CODE REVIEW

**Comment by m4tt075** [comment]
@ymartin59 Gave it another try and got it working, but please review...

File: cross/ffmpeg/Makefile [hunk 1]
```diff
@@ -42,6 +42,10 @@ ifeq ($(findstring $(ARCH),$(ARM7_ARCHES)),$(ARCH))
 CONFIGURE_ARGS += --arch=arm --disable-neon --disable-armv6 --disable-armv6t2 --disable-vfp --disable-armv5te --disable-yasm --disable-asm
 endif
 
+ifeq ($(findstring $(ARCH),$(ARM8_ARCHES)),$(ARCH))
+CONFIGURE_ARGS += --arch=arm64 --disable-neon --disable-yasm --disable-asm
```
[the remark this answers is not in the document]
Comment (ymartin59) [reply cross/ffmpeg/Makefile]: According to https://developer.arm.com/technologies/neon we should consider NEON is available for both ARMv7 and ARMv8.
I propose to give a try to remove `--disable-neon` flag for these two architecture.
May you please build for `rtd1297` this PR and submit to requester for "beta" testing ?

**Comment by ymartin59** [comment]
Reference to requesters: #2997 #3083 #2886 #2981 #3079

**Comment by ymartin59** [comment]
I have built and submitted package for testing

**Comment by m4tt075** [comment]
Yes, makes sense. Just pushed the changes. Building [here](https://travis-ci.org/m4tt075/spksrc/builds/331409037).

**Comment by Idree** [comment]
Hello, i am betatesting your package :)

I just installed it on my DS216play which is packed with an rtd1296 and video's with DTS seems to be working.

Amazing work 👍

**Comment by Safihre** [comment]
@m4tt075 any idea why all the `x86/64` toolchains fail? Can't really read the log well here on mobile.

**Comment by m4tt075** [comment]
@Safihre Yes, found it. Working on it. Pushing update soon...

**Comment by m4tt075** [comment]
Next build attempts [here](https://travis-ci.org/m4tt075/spksrc/builds/331429593)

**Comment by ymartin59** [comment]
Already one user reported `rtd1296` package works well

**Comment by ymartin59** [comment]
I have to review my pending branches with some updates related to `ffmpeg`, `mediainfo` and `chromaprint`

**Comment by m4tt075** [comment]
OK, all 6.1 builds are green. Test packages are [here](https://github.com/m4tt075/spksrc/releases/tag/ffmpeg-3.3.3-arm).

@ymartin59 Feel free to cherry-pick my commits into a bigger `ffmpeg` update PR if you like to. I will be AFK for most of next week. And I know there is more we could do (update to 3.4.1, hw acceleration support for transcoding, specific patches which are only applied when it is compiled as part of TVH).

**Comment by ymartin59** [comment]
@m4tt075 Thanks a lot. That is my idea. I take opportunity to update as much as possible because of publication process cost...

**Comment by Safihre** [comment]
@ymartin59 why not use the (Docker based) Travis image to do the publishing? You can add your SynoCommunity login details as a Travis secret/encrypted environment variable. 
This is also how we do it for the SABnzbd releases.

**Comment by ymartin59** [comment]
Replaced by #3136. Unfortunately no longer builds for DSM 5.2

**Pull request closed without merging**